include/tvm/relay/expr_functor.h

```diff
@@ -196,7 +196,7 @@ class ExprMutator
    * \brief Mutate is alias for VisitExpr
    * \return expr.
    */
-  Expr Mutate(const Expr& expr) {
+  virtual Expr Mutate(const Expr& expr) {
```

Comment:
I still think it is better to not subclass Mutate, instead, override VisitExpr in the ScopeMutator, which calls into DispatchVisitExpr that does the dispatching.

This way we do not have to change a lot of calls of VisitExpr into Mutate in the subclasses, which can be confusing(for user to think about which one to call).